### PR TITLE
1421 checkbox migration

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
IcCheckbox/IcCheckboxGroup cypress migration including a11y and visual regression tests.

## Related issue
#1421 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

File: packages/react/cypress-visual-screenshots/baseline/IcCheckbox.cy.tsx-default.png
Comment (GCHQ-Developer-299) [inline packages/react/cypress-visual-screenshots/baseline/IcCheckbox.cy.tsx-default.png:1]: this screenshot is empty - unecessary?

Comment (gd2910) [reply packages/react/cypress-visual-screenshots/baseline/IcCheckbox.cy.tsx-default.png:1]: The HTML tags for Checkbox went wandering, fixed now :)

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Comment by GCHQ-Developer-465** [comment]
A rebase is needed - looks great though :)

File: packages/react/src/component-tests/IcCheckbox/IcCheckboxTestData.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          <IcTextField slot="additional-field" label="Please let us know..." />
+        </IcCheckbox>
+      </IcCheckboxGroup>
+    </div>
+  );
+};
+
+export const CheckboxForm = () => {
+  const [submit, setSubmit] = useState("");
+  const handleSubmit = (event) => {
```
Comment (GCHQ-Developer-847) [inline packages/react/src/component-tests/IcCheckbox/IcCheckboxTestData.tsx:36]: Does `event` need a type? Getting this error / warning about it when running the Cypress tests:

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Pull request merged**